### PR TITLE
feat: support SQLite AUTOINCREMENT column attribute

### PR DESCRIPTION
**Pull request by tsuty**
I implemented for 'AUTOINCREMENT' column attribute syntax for SQLite. 

I thought that 'ColumnType' should need another `Autoincrement` field for sqlite, or 'ParserMode'.
This implementation is the latter.

### CODE REVIEW

File: sqlparser/parser.y [hunk 1]
```diff
@@ -789,6 +789,12 @@ column_definition_type:
     $1.Autoincrement = BoolVal(true)
     $$ = $1
   }
+| column_definition_type AUTOINCREMENT
+  {
+    $1.Autoincrement = BoolVal(true)
+    $1.ParserMode = ParserModeSQLite3
```
Comment (k0kubun) [inline sqlparser/parser.y:795]: `ParserMode` doesn't feel like an attribute of ColumnType. WDYT about adding `AutoIncrement`, using it for `AUTO_INCREMENT`, and using `Autoincrement` for `AUTOINCREMENT` instead?

Comment (k0kubun) [reply sqlparser/parser.y:795]: Hmm, but then other implementations would need to deal with both. Let me reconsider it.

**Comment by k0kubun** [comment]
> I thought that 'ColumnType' should need another Autoincrement field for sqlite, or 'ParserMode'.
This implementation is the latter.

This seems to be used only for `*ColumnType.Format()`. Is the `Format()` method used by sqldef? If not, we might be able to simply skip both of them.

Also, it'd be nice if you can add tests to `cmd/sqlite3def/sqlite3def_test.go` instead. We don't really maintain `sqlparser/parse_test.go`. If the test is the only reason to maintain the `ParserMode`, it's fine to just remove the tests.

**Comment by tsuty** [comment]
> This seems to be used only for *ColumnType.Format(). Is the Format() method used by sqldef? If not, we might be able to simply skip both of them.

sqldef does not use it. 

> We don't really maintain sqlparser/parse_test.go If the test is the only reason to maintain the ParserMode, it's fine to just remove the tests.

I understand. implement it.
I am only aiming for no syntax error for sqlite3def command.
And I just noticed that need implement for sqlite internal tables 
 (sqlite_sequence...) 

So, I need more  imprements.

**Comment by k0kubun** [comment]
Please let me know when your next patch becomes ready. I'm closing this for now since I can't merge this PR due to conflicts.

**Pull request closed without merging**